pkg/controller/v1alpha1/trainedmodel/controller.go

```diff
@@ -118,15 +122,78 @@ func (r *TrainedModelReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		return ctrl.Result{}, nil
 	}
 
+	// update URL and Address fo TrainedModel
+	if err := r.updateStatus(req, tm); err != nil {
+		return ctrl.Result{}, err
+	}
+
 	// Reconcile modelconfig to add this TrainedModel to its parent InferenceService's configmap
 	if err := r.ModelConfigReconciler.Reconcile(req, tm); err != nil {
 		return ctrl.Result{}, err
 	}
 	return ctrl.Result{}, nil
 }
 
-func (r *TrainedModelReconciler) updateStatus(desiredService *v1alpha1api.TrainedModel) error {
-	//TODO update TrainedModel status object, this will be done in a separate PR
+func (r *TrainedModelReconciler) updateStatus(req ctrl.Request, desiredModel *v1alpha1api.TrainedModel) error {
+	// Get the parent inference service
+	isvc := &v1beta1api.InferenceService{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Namespace: req.Namespace, Name: desiredModel.Spec.InferenceService}, isvc); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Parent InferenceService does not exists, deleting TrainedModel", "TrainedModel", desiredModel.Name, "InferenceService", isvc.Name)
+			r.Delete(context.TODO(), desiredModel)
```

Comment:
I think we shouldn't delete trained model here, instead we can return error back and the next reconciliation loop will retry.